server/service/src/sync/test/integration/remote/mod.rs

```diff
@@ -74,8 +74,10 @@ async fn test_remote_sync_record(identifier: &str, tester: &dyn SyncRecordTester
         replace_system_name_ids(&mut integration_records, &previous_connection);
 
         // Integrate
-
         integration_records.integrate(&previous_connection).unwrap();
+
+        replace_is_sync_update(&mut integration_records);
```

Comment:
I would move it down after the sync/push because after the re-initialization you expect `is_sync_update = true` when validating the pulled records